### ISSUE TITLE
Issue Exporting YOLO11 Model to TFLite with INT8 Quantization

### ISSUE THREAD

**Issue by AlaaArboun** [ultralytics/ultralytics#19326]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi everyone,

I'm encountering an issue when exporting a YOLO11 model to TFLite with INT8 quantization. I'm using the following code:

```python
from ultralytics import YOLO

model = YOLO('yolo11s.pt')
results = model.export(format='tflite', task="detect", int8=True, imgsz=224, data="coco.yaml")
```

The export fails, and I'm not sure what might be causing the problem. Has anyone experienced this issue before? Specifically, I'm wondering:

Is INT8 quantization fully supported for the YOLO11 export?
Are there any known workarounds or additional settings required?
Could this be related to specific versions of the Ultralytics library or the model file?
I've already tried updating to the latest version, but the error still persists. Any insights or suggestions would be greatly appreciated.

Thanks in advance!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AlaaArboun, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. From your code snippet, the issue seems to relate to exporting YOLO11 to TFLite with INT8 quantization. Please include:

- The full error message or traceback you are encountering.
- Your Python, PyTorch, and `ultralytics` versions (these can be printed with `pip show ultralytics` and `torch.__version__`).
- Your system environment details (e.g., OS, GPU/CPU setup).
- Whether you used custom modifications to the model or dataset (if so, include relevant configurations).

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Ensure you are running the latest `ultralytics` package which supports YOLO11 and includes all required dependencies. Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO models may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response 🦾, but no worries! An Ultralytics engineer will follow up to assist you further. If you can provide the requested details, it will help us resolve your issue more efficiently.

**Comment by Y-T-G** [comment]
"coco.yaml" would download the 118k images for calibration. You should be using a smaller dataset for calibration. Like "coco128.yaml"

**Comment by glenn-jocher** [comment]
Thank you for sharing your implementation approach. For streamlined TFLite INT8 export with YOLO11 models, we recommend using the native Ultralytics export functionality which handles output formatting automatically. The [TFLite export documentation](https://docs.ultralytics.com/integrations/tflite) provides specific guidance for INT8 quantization using our calibration implementation. 

If you encounter specific issues with the native export path, please share error details and we'll help troubleshoot. For third-party conversion workflows like Nobuco, we recommend reaching out to their maintainers for framework-specific support.
